src/components/views/settings/SetIdServer.js

```diff
@@ -0,0 +1,243 @@
+            // probably want a /versions endpoint like the C/S API.
+            { method: "GET", url: isUrl + '/_matrix/identity/api/v1' },
+            (err, response, body) => {
+                if (err) {
+                    resolve(_t("Could not connect to ID Server"));
+                } else if (response.status < 200 || response.status >= 300) {
+                    resolve(_t("Not a valid ID Server (status code %(code)s)", {code: response.status}));
+                } else {
+                    resolve(null);
+                }
+            },
+        );
+    });
+}
+
+export default class SetIdServer extends React.Component {
+    constructor() {
+        super();
+
+        let defaultIdServer = abbreviateUrl(MatrixClientPeg.get().getIdentityServerUrl());
+        if (!defaultIdServer) {
+            defaultIdServer = abbreviateUrl(SdkConfig.get()['validated_server_config']['idServer']) || '';
+        }
+
+        this.state = {
+            currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
+            idServer: defaultIdServer,
+            error: null,
+            busy: false,
+        };
+    }
+
+    _onIdentityServerChanged = (ev) => {
+        const u = ev.target.value;
+
+        this.setState({idServer: u});
+    };
+
+    _getTooltip = () => {
+        if (this.state.busy) {
+            const InlineSpinner = sdk.getComponent('views.elements.InlineSpinner');
+            return <div>
+                <InlineSpinner />
+                { _t("Checking Server") }
+            </div>;
+        } else if (this.state.error) {
+            return this.state.error;
+        } else {
+            return null;
+        }
+    };
+
+    _idServerChangeEnabled = () => {
+        return !!this.state.idServer && !this.state.busy;
+    };
+
+    _saveIdServer = async () => {
+        this.setState({busy: true});
+
+        const fullUrl = unabbreviateUrl(this.state.idServer);
+
+        const errStr = await checkIsUrl(fullUrl);
+
+        let newFormValue = this.state.idServer;
+        if (!errStr) {
+            MatrixClientPeg.get().setIdentityServerUrl(fullUrl);
+            localStorage.removeItem("mx_is_access_token");
+            localStorage.setItem("mx_is_url", fullUrl);
+            newFormValue = '';
+        }
+        this.setState({
+            busy: false,
+            error: errStr,
+            currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
+            idServer: newFormValue,
+        });
+    };
+
+    _onDisconnectClicked = () => {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        Modal.createTrackedDialog('ID Server Disconnect Warning', '', QuestionDialog, {
+            title: _t("Disconnect ID Server"),
+            description:
+                <div>
+                    {_t(
+                        "Disconnect from the ID Server <idserver></idserver>?", {},
```

Comment:
I believe our i18n lets you use `<idserver/>`. Alternatively: `<b>%(idserver)s</b>` then do a replacement on `b`.